hibernate-core/src/main/java/org/hibernate/type/descriptor/java/SerializableTypeDescriptor.java

```diff
@@ -45,14 +44,16 @@ public S deepCopyNotNull(S value) {
 
 	}
 
-	@SuppressWarnings({ "unchecked" })
 	public SerializableTypeDescriptor(Class<T> type) {
-		super(
-				type,
-				Serializable.class.equals( type )
-						? (MutabilityPlan<T>) SerializableMutabilityPlan.INSTANCE
-						: new SerializableMutabilityPlan<T>( type )
-		);
+		super( type, createMutabiltyPlan( type ) );
+	}
+
+	@SuppressWarnings({ "unchecked" })
+	private static <T> MutabilityPlan<T> createMutabiltyPlan(Class<T> type) {
```

Comment:
Love this idea of extending `@Immutable` to attributes and interpreting that in terms of `MutabilityPlan`.  One place you missed for that is in `org.hibernate.type.descriptor.java.JavaTypeDescriptorRegistry.FallbackJavaTypeDescriptor`.  Could you add this logic there as well?  Thanks